### ISSUE TITLE
In Key Test Report Features, define "expected" and "actual" test results

### ISSUE THREAD

**Issue by ironprogrammer** [WordPress/test-handbook#2]
This update pertains to the [Key Test Report Features section](https://make.wordpress.org/test/handbook/test-reports/#key-test-report-features) and follows the initial publication of the Test Reports page, [discussed in Make Test Slack](https://wordpress.slack.com/archives/C03B0H5J0/p1654800411855519?thread_ts=1654798408.140869&cid=C03B0H5J0). From the discussion:

> I would also suggest to divide Test Results in the Key Test Report Features in to 2 parts (though it is being mentioned later in the article):
> - Expected result
> - Actual result

This will be especially helpful once #1 has been resolved.

**Comment by SirLouen** [comment]
We have been commenting this in todays Testing program session

The problem with "Expected" results is that the tester should not have any expectations. 

Expectations should be set by the reporter. 

Maybe the tester could fetch the expectations and also write them up in the report, just to double check if the expected result aligns with the reporter expectation. But since adding expectations by reporters is not required (and for example in Gutenberg, is not even a field during the bug report creation process), this is forcing testers to add expectations when there are no clear expectations in most of the patches or reports. I have already reported this to Gutenberg without any success. 

Also the test results or the actual results are totally redundant. When going through the steps the last step is the result itself. Example:

```
=== Steps taken
1.
2.
3.
x. 🐞 Bug occurs / ❌ Bug is not occurring
```
But the actual result is pretty obvious because we are clearly showcasing it through the steps, so its totally redundant.

**Comment by ironprogrammer** [comment]
Hi, @SirLouen 👋🏻 The common lack of documented expectations for a given bug was the underlying reason why this was absorbed into a Test Team function.

You're right, it's not ideal, but adding expectations attempts to address that gap, even if it requires someone other than the reporter (which can really be any contributor) to perform some research, and/or collaborate with the reporter in order to add these to the ticket. During my tenure with the team, this was a [_mise en place_](https://en.wikipedia.org/wiki/Mise_en_place) required before performing actual tests. The [`reporter-feedback`](https://core.trac.wordpress.org/query?status=accepted&status=assigned&status=new&status=reopened&status=reviewing&keywords=~reporter-feedback&col=id&col=summary&col=status&col=owner&col=type&col=priority&col=milestone&order=priority) flag often goes ignored.

Off the top of my head I cannot recall specific attempts around _requiring_ this info on bug reports (in Trac or GH), but I seem to recall the general pushback is that fewer things are reported when there are more barriers put in place. Each bug report should be seen as a collaborative effort to extract the details, and sometimes the reporter does a fine job, and other times it requires others from the community.

**Comment by SirLouen** [comment]
Hey @ironprogrammer yeah, I know that many users have like the most enlightened pushback ideas for every single topic that comes down to the table about testing. 

The reality is that nowadays the bug reports and the commits done look something like this

<img width="1122" height="549" alt="Image" src="https://github.com/user-attachments/assets/1147aba3-af10-403d-a12c-86a8eb19b790" />

I've been mocking this fact for a year and I know that some people dont like me to do this mock but is a fact. So raising a bit the standards seems to be necessary to actually solve the issues. 95% of the tickets are abandoned and will never receive more followup from the reporters, which means, that they are useless bugs as we will not be able to fix them without that extra feedback.

In fact, in many issue trackers, they have this one-three-six-month stale policy, where the tickets autoclose if they dont have any answers (this is a good scale to understand on average how much OSS value the bug reports). I know that here there are some sybarites that think that "each bug report is a precious gem". 95% of those gems are already rotten.

Having these premises in mind, basically we were talking yesterday about one of the most problematic things with bug reports: There is no placeholder for reports in Core as they are Gutenberg. Even in Gutenberg, the "expected" thing is not even mandatory. But I'm not going to "force" core to implement this. We will stick to play with the toys we have around. At this point one of the "core targets of the Test Team", is to drop `needs-testing` tickets down to 0 before 2027. We will see afterwards.

So basically we have concluded, that we may ultimately put in the reports, what do we think the user expects at least to have a reference and if the reporter reads it they can say: "Hey I was not expecting that!" . 

Now we are all playing around with the two #104 experimental reports to see how it feels this new format we somewhat agreed.

**Comment by huzaifaalmesbah** [comment]
Thanks for opening this issue. We have introduced new report templates in #104 (merged via #166) that now include dedicated **Steps taken** and **Expected result** sections. Closing this as resolved.
